docs/src/main/asciidoc/health-guide.adoc

```diff
@@ -0,0 +1,227 @@
+
+import javax.enterprise.context.ApplicationScoped;
+
+@Health
+@ApplicationScoped
+public class DataHealthCheck implements HealthCheck {
+
+    @Override
+    public HealthCheckResponse call() {
+        return HealthCheckResponse.named("Health check with data")
+                .up()
+                .withData("foo", "fooValue")
+                .withData("bar", "barValue")
+                .build();
+    }
+}
+----
+
+If you rerun the health check procedure again by accessing `/health` endpoint you can 
+see that the new health check `Health check with data` is present in `checks` array. 
+This check contains a new attribute called `data` which is a JSON object consisting of 
+properties we have defined in our health check procedure.
+
+== Negative health check procedure
+
+In this section we create another health check procedure which simulates connection to 
+external service provider like database. For the simplicity reasons, we only determine 
+whether the database is accessible or not by a configuration property.
+
+Create `org.acme.health.DatabaseConnectionHealthCheck` class:
+
+[source,java]
+----
+package org.acme.health;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.eclipse.microprofile.health.Health;
+import org.eclipse.microprofile.health.HealthCheck;
+import org.eclipse.microprofile.health.HealthCheckResponse;
+import org.eclipse.microprofile.health.HealthCheckResponseBuilder;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+@Health
+@ApplicationScoped
+public class DatabaseConnectionHealthCheck implements HealthCheck {
+    
+    @Inject
+    @ConfigProperty(name = "database.up", defaultValue = "false")
+    private boolean databaseUp;
+    
+    @Override
+    public HealthCheckResponse call() {
+        
+        HealthCheckResponseBuilder responseBuilder = HealthCheckResponse.named("Database connection health check");
+
+        try {
+            simulateDatabaseConnectionVerification();
+            responseBuilder.up();
+        } catch (IllegalStateException e) {
+            // cannot access the database
+            responseBuilder.down()
+                    .withData("error", e.getMessage());
+        }
+
+        return responseBuilder.build();
+    }
+
+    private void simulateDatabaseConnectionVerification() {
+        if (!databaseUp) {
+            throw new IllegalStateException("Cannot contact database");
+        }
+    }
+}
+----
+
+If you now rerun the health check the overall `outcome` should be DOWN and you should 
+see in the `checks` array the newly added `Database connection health check` which is 
+down and the error message explaining why it failed.
+
+As we shouldn't leave this application with health check in DOWN state and because we 
+are running {project-name} dev mode you can add `database.up=true` in 
+`META-INF/microprofile-config.properties` and rerun the health check again -- 
```

Comment:
```suggestion
`src/main/resources/application.properties` and rerun the health check again -- 
```
This might need a change to the quickstart itself if you included the configuration file.